tests/Generator.hs

```diff
@@ -0,0 +1,82 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE FlexibleInstances #-}
+
+module Generator where
+
+import Test.Tasty
+import Test.QuickCheck.Instances ()
+import Test.Tasty.HUnit ((@=?), testCase)
+import Test.Tasty.Golden (goldenVsStringDiff)
+import Control.Lens (over, (&), set, at)
+import qualified Data.Text.Encoding as T
+import qualified Data.ByteString.Lazy as LB
+import Data.Time.Clock (UTCTime(..), secondsToDiffTime)
+import Data.Time.Calendar (Day(..))
+
+import qualified Data.CaseInsensitive as CI
+
+import Data.MIME
+
+properties :: TestTree
+properties =
+    testGroup
+        "message rendering tests"
+        [ renderSimpleMailSuccessfully
+        , rendersMultiPartSuccessfully
+        , exampleMailsParseSuccessfully
+        ]
+
+renderSimpleMailSuccessfully :: TestTree
+renderSimpleMailSuccessfully =
+    goldenVsStringDiff
+        "renders simple mail"
+        diffCommand
+        "tests/golden/textplain7bit.golden"
+        (pure $ LB.fromStrict $ renderMessage textPlain7bit)
+
+diffCommand :: FilePath -> FilePath -> [String]
+diffCommand ref new =
+    [ "diff"
+    , "-d" -- minimal
+    , "-u" -- unified
```

Comment:
Why not just use the long options? (They are the same on BSD and GNU).

Reply:
Heh true. Which are exactly what my comments are. Doh.